build/builder.go

```diff
@@ -655,6 +674,13 @@ func (b *Builder) Finish() error {
 				}
 			}
 
+			if b.opts.GetHash() != repository.IndexOptions {
+				return &deltaIndexOptionsMismatchError{
+					shardName:  shard,
+					newOptions: b.opts.HashOptions(),
+				}
+			}
```

Comment:
Thanks @keegancsmith , I'd still like to meet to discuss the stylistic suggestions once the red-accounts effort wraps up, since I may not understand the comments above re: logging